compiler/rustc_ast_passes/src/ast_validation.rs

```diff
@@ -219,10 +219,30 @@ impl<'a> AstValidator<'a> {
                     }
                 }
             }
+            TyKind::AnonymousStruct(ref fields, ..) | TyKind::AnonymousUnion(ref fields, ..) => {
+                // self.with_banned_assoc_ty_bound(|this| {
+                walk_list!(self, visit_struct_field_def, fields)
+                // });
+            }
             _ => visit::walk_ty(self, t),
         }
     }
 
+    fn visit_struct_field_def(&mut self, field: &'a FieldDef) {
+        if let Some(ident) = field.ident {
+            if ident.name == kw::Underscore {
+                self.check_anonymous_field(field);
+                self.visit_vis(&field.vis);
+                self.visit_ident(ident);
+                self.visit_ty_common(&field.ty);
+                self.walk_ty(&field.ty);
+                walk_list!(self, visit_attribute, &field.attrs);
+                return;
+            }
+        }
```

Comment:
Another minor nit/food for thought, but perhaps worth considering using a let chain here  instead or adjusting the pattern to be able to avoid the extra level of indentation